common.go

```diff
@@ -0,0 +1,37 @@
+package set
+
+// CommonSet is the interface that all sets implement
+type CommonSet[T any] interface {
+	Slice() []T
+	Insert(T) bool
+	InsertSlice([]T) bool
+	Size() int
+	// ForEach  will call the callback function for each element in the set
+	// If the callback returns false, the iteration will stop
```

Comment:
Let's also add a note that iteration order depends on the underlying implementation